language/move-stdlib/sources/Capability.move

```diff
@@ -100,6 +112,19 @@ module Std::Capability {
     /// can succeed with this operation. A caller must pass a witness that they own the `Feature` type
     /// parameter.
     public fun acquire<Feature>(requester: &signer, _feature_witness: &Feature): Cap<Feature>
+    acquires CapState, CapDelegateState {
+        Cap<Feature>{root: validate_acquire<Feature>(requester)}
+    }
+
+    /// Acquires a linear capability token. It is up to the module which owns `Feature` to decide
+    /// whether to expose a linear or non-linear capability.
+    public fun acquire_linear<Feature>(requester: &signer, _feature_witness: &Feature): LinearCap<Feature>
+    acquires CapState, CapDelegateState {
+        LinearCap<Feature>{root: validate_acquire<Feature>(requester)}
```

Comment:
Should this remove `CapDelegateState` from the `requester` (if applicable)? If we don't to this, the delegate can acquire multiple copies of the allegedly linear capability which would be bad if the intention is to (e.g.) let the delegate spend $10 once.

Reply:
Again, this would going to be handled in the module which owns `Feature` and provides the `acquire` (and the probably also a release). But notice that you can anyway only get LinearCap if you have a signer. So you would typically acquire this capability somewhere once at the top-level of a transaction, then don't pass the signer but the cap to called functions, and the cap would have the desired linearity properties.